src/main/java/swervelib/SwerveDrive.java

```diff
@@ -1042,12 +1023,7 @@ public void setGyroOffset(Rotation3d offset) {
   public void addVisionMeasurement(Pose2d robotPose, double timestamp) {
     odometryLock.lock();
     swerveDrivePoseEstimator.addVisionMeasurement(robotPose, timestamp);
-    Pose2d newOdometry = new Pose2d(swerveDrivePoseEstimator.getEstimatedPosition().getTranslation(),
-        robotPose.getRotation());
     odometryLock.unlock();
-
-    // setGyroOffset(new Rotation3d(0, 0, robotPose.getRotation().getRadians()));
-    resetOdometry(newOdometry);
```

Comment:
Removed this call, as is calls:
 - `swerveDrivePoseEstimator.resetPosition`
 - which calls `PoseEstimator.resetPosition`
 - which calls `m_poseBuffer.clear`, which is the time buffer whose size we were trying to test reducing from 1.5 secs to something smaller.

We are essentially not letting this buffer grow and gather data for more than 20 or so ms, as we reset it with each vision measurement. I suspect this leads to our jittery pose estimation when vision is enabled.

I think in all our discussions about the std dev and how the kalman filter works - I have never quite seen it work reliably with vision enabled. I think this is why our pose is so much more stable when vision is off, but this is subject to be tested and confirmed.